src/validation.cpp

```diff
@@ -1858,12 +1858,58 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
     // before the first had been spent.  Since those coinbases are sufficiently buried its no longer possible to create further
     // duplicate transactions descending from the known pairs either.
     // If we're on the known chain at height greater than where BIP34 activated, we can save the db accesses needed for the BIP30 check.
+
+    // The exception to this logic is if there are coinbases from blocks before
+    // the BIP34 height of 227,931 which have a scriptSig with a valid-looking
+    // BIP34 height for a later block height. These coinbases run the risk of
+    // being duplicated when we reach that later block height. An exhaustive
+    // search of all mainnet coinbases before the BIP34 height which have a
+    // scriptSig indicating a later block height reveals 2 cases where the later
+    // block height is before 1,983,702.
```

Comment:
Is the significance of 1,938,702 just that it is the third lowest height found encoded in the old coinbases? The sentence doesn't say where 1,938,702 comes from, and this is written as if it should be obvious what the number means.

If it is just the third lowest height found in the search, I think it would be clearer to say something like "An exhaustive search of all blocks before the BIP34 height found that the lowest height values indicated in coinbase scriptSigs were: 209,921, 490,897, and 1,983,702."